### DISCUSSION TITLE
Monitor upstream fork

### DISCUSSION THREAD

**Discussion by micah** [renovatebot/renovate#12159] [Suggest an Idea]
I would like a way for renovate to monitor an 'upstream' project for changes, and to propose merging (or fast-forwarding) changes that appear upstream into my repository. 

Renovate does a really good job at checking if there are newer versions which can be updated to, but there are some repositories who do not do releases, and simply make commits and I'd like renovate to track those commits and notify me with a merge request when they have been made.

Those projects are not submodules, they are just distinct repositories located on external repositories (be they gitlab, github, or self-hosted git repositories).

I haven't found a way to do this with renovate, but perhaps I am missing how it is referenced? Is there a way to do this now, or should I file a feature request?

**Comment by rarkins** [comment]
What type of file do you reference them in, in your downstream repository?

**Reply by micah** [reply]
For example, if I am interested in tracking a repository from github, lets take the renovate repository as an example. I would clone it locally:

`git clone https://github.com/renovatebot/renovate.git`

then I would like to host this repository at gitlab, where I'm going to make some minor modifications, but nothing significant that is worth making a MR/PR to bring it upstream. I'd make a new project at gitlab, and then I would add my gitlab project as a remote in the existing clone of the renovate repository:

`cd renovate`
`git remote add gitlab https://gitlab.com/myspace/renovatebot/renovate.git`
`git push gitlab`

This is just basic git work flow here. There is no reference to the githb version of the repository in my fork on gitlab, its only in my local git repository's config as a remote.

The typical workflow here would be I would need to periodically go into this directory and run `git remote update` and see there are changes in the github version of the repository, and then work on merging those and then push them to gitlab.

For an actively developed project, this is not a big deal, I don't mind doing that in the local git repository. I want to stay up-to-date, and rebase to stay current. 

However, if this is not an actively developed repository, just something that gets changes every few months or so because the github repository is in maintenance mode (maybe when a library dependency has been updated or something), then I'm not going to regularly change directory into my local repository to run the `git remote update`, and this is where renovate would be super helpful, because it could check on these things and let me know to pay attention after a few months.

A way to track this would be just to have a 'remote' git repository option that you would add to renovate.json, so renovate knows to go look at it and compare it with what it is monitoring.

**Reply by rarkins** [reply]
How would the workflow look like? e.g. can Renovate somehow create a Pull Request for this, and would you want it automatically merged or you'd approve it?

**Reply by micah** [reply]
That is a good question. I'm guessing renovate could act like it does now, it creates a pull request with the changes, and be configured, like it can now, with the optional `automerge` based on whatever defined criteria.

**Comment by DanySK** [comment]
This feature would be great for me and my students too. We often fork off repositories and write additional components, and having a way to keep the repository up to date (possibly, via rebasing) would be great.

**Reply by rarkins** [reply]
@DanySK do you just mean keeping your own fork in sync? If you're using GitHub, they made it very easy via a single click now at least

**Reply by DanySK** [reply]
Yes, but one click is one click too much for me :wink: 
I usually configure hooks in git to rebase pull upstream after committing, but it might be useful to automate the process when a forked repo is only rarely edited (maybe, hosts a minimally patched version of some product)

**Reply by viceice** [reply]
I think we had a feature request like this some time before and decided that this isn't something which can be easily implemented in current renovate.

Maybe there is an open issue for this and we wait for more users requesting this feature.

**Reply by rarkins** [reply]
Yes, because it's not possible to do with Pull Requests so doesn't fit into our existing model

**Reply by micah** [reply]
For me the problem is that I'm often not working on github itself. I'm working on gitlab, or a project is on a self-hosted gitlab, or other repository hosting solution. So while it is nice that github has this nice feature, its not something that is available everywhere.
